### PR TITLE
chore: reformat pull request template

### PR DESCRIPTION
**Pull request by gracelu0**
### Reason for this change

I noticed that some PRs only link the issue or have really long paragraph descriptions which can be hard to follow and understand the why/what regarding the changes.

### Description of changes

Reformatting the pull request template with section headings. This enforces PR descriptions to follow a uniform format and include the information we want. Should make it easier/more efficient to review PRs. 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by GavinZZ — COMMENTED** [review]
nit: I really liked SAM-Transform repo's PR template where we issues as a section and users can always add more detail to the issue section to explain the issue.
```
### Issue #, if available
Closes #<issue number here>.
```

Maybe make the checklist its own section and we can append to it in the future
```
### Checklist
- [ ] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and the [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)
```

**Comment by gracelu0** [comment]
> nit: I really liked SAM-Transform repo's PR template where we issues as a section and users can always add more detail to the issue section to explain the issue.
> 
> ```
> ### Issue #, if available
> Closes #<issue number here>.
> ```
> 
> Maybe make the checklist its own section and we can append to it in the future
> 
> ```
> ### Checklist
> - [ ] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and the [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)
> ```

I think having the issue at the top is a good idea. I can add a `Checklist` heading as well to keep it organized. I believe the entire template used to be a long checklist but it got removed since people weren't really using it - something to keep in mind if before we decide to add more checklist items.

**Comment by GavinZZ** [comment]
>I believe the entire template used to be a long checklist but it got removed since people weren't really using it - something to keep in mind before we decide to add more checklist items.

That's a good point actually, I tend to ignore all checklist most of the time as well. Maybe we could have it instead as a `USEFUL LINKS` section if we find it not useful in the future and just put a couple of useful links CDK has.

**Review by GavinZZ — APPROVED** [review]

**Comment by GavinZZ** [comment]
@Mergify update

**Comment by paulhcsun** [comment]
@mergifyio update

**Comment by moelasmar** [comment]
do you think we should add a section for the Breaking changes as mentioned [here](https://github.com/aws/aws-cdk/blob/b32f47cfd65a5288e7bc2cc5a8a1f8e5f59032f8/CONTRIBUTING.md?plain=1#L620)?

**Comment by gracelu0** [comment]
> do you think we should add a section for the Breaking changes as mentioned [here](https://github.com/aws/aws-cdk/blob/b32f47cfd65a5288e7bc2cc5a8a1f8e5f59032f8/CONTRIBUTING.md?plain=1#L620)?

I think that's a good idea since there's a specific format required, I can add it
